### PR TITLE
Additional bench tracing improvements

### PR DESCRIPTION
**Pull request by timneutkens**
- Updates launch.json as `test/integration/basic` is removed
- Added additional scripts
- Uses direct webpack5 instead of the bundled one
- Uses a faster traceId generator

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes

### CODE REVIEW

File: packages/next/trace/trace.ts [hunk 1]
```diff
@@ -1,10 +1,9 @@
-import { randomBytes } from 'crypto'
-import { SpanId } from './shared'
+import { customAlphabet } from 'next/dist/compiled/nanoid/index.cjs'
 import { reporter } from './report'
 
 const NUM_OF_MICROSEC_IN_SEC = BigInt('1000')
-
-const getId = () => randomBytes(8).toString('hex')
+const nanoid = customAlphabet('0123456789abcdefghijklmnopqrstuvwxyz', 8)
+const getId = () => Buffer.from(nanoid(), 'utf8').toString('hex')
```
Comment (sokra) [inline packages/next/trace/trace.ts:6]:
```suggestion
const nanoid = customAlphabet('0123456789abcdef', 16)
const getId = () => nanoid()
```

**Review by sokra — CHANGES_REQUESTED** [review]

File: packages/next/trace/trace.ts (hunk 1 above)
Comment (timneutkens) [reply packages/next/trace/trace.ts:6]: Using nanoid does not work because it includes - and _ as well as capitals which is not supported by jaeger.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| buildDuration | 14.2s | 15.6s | ⚠️ +1.4s |
| buildDurationCached | 3.5s | 3.6s | ⚠️ +142ms |
| nodeModulesSize | 182 MB | 182 MB | ⚠️ +230 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.102 | 3.518 | ⚠️ +0.42 |
| / avg req/sec | 805.83 | 710.55 | ⚠️ -95.28 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.752 | 1.963 | ⚠️ +0.21 |
| /error-in-render avg req/sec | 1427.23 | 1273.74 | ⚠️ -153.49 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| 779.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 27 kB | 27 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 70.8 kB | 70.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| _app-HASH.js gzip | 977 B | 977 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 311 B | 311 B | ✓ |
| css-HASH.js gzip | 328 B | 328 B | ✓ |
| dynamic-HASH.js gzip | 2.67 kB | 2.67 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 918 B | 918 B | ✓ |
| image-HASH.js gzip | 4.15 kB | 4.15 kB | ✓ |
| index-HASH.js gzip | 260 B | 260 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 13 kB | 13 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| _buildManifest.js gzip | 494 B | 494 B | ✓ |
| Overall change | 494 B | 494 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| index.html gzip | 538 B | 538 B | ✓ |
| link.html gzip | 550 B | 550 B | ✓ |
| withRouter.html gzip | 532 B | 532 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| buildDuration | 7.2s | 7.3s | ⚠️ +115ms |
| buildDurationCached | 3.5s | 3.6s | ⚠️ +140ms |
| nodeModulesSize | 182 MB | 182 MB | ⚠️ +230 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.253 | 3.266 | ⚠️ +0.01 |
| / avg req/sec | 768.42 | 765.36 | ⚠️ -3.06 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.765 | 1.886 | ⚠️ +0.12 |
| /error-in-render avg req/sec | 1416.36 | 1325.57 | ⚠️ -90.79 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| 675-HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| 770.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 50.7 kB | 50.7 kB | ✓ |
| main-HASH.js gzip | 34.9 kB | 34.9 kB | ✓ |
| webpack-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| Overall change | 101 kB | 101 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.32 kB | 1.32 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 315 B | 315 B | ✓ |
| css-HASH.js gzip | 331 B | 331 B | ✓ |
| dynamic-HASH.js gzip | 2.8 kB | 2.8 kB | ✓ |
| head-HASH.js gzip | 356 B | 356 B | ✓ |
| hooks-HASH.js gzip | 637 B | 637 B | ✓ |
| image-HASH.js gzip | 573 B | 573 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 2.2 kB | 2.2 kB | ✓ |
| routerDirect..HASH.js gzip | 326 B | 326 B | ✓ |
| script-HASH.js gzip | 393 B | 393 B | ✓ |
| withRouter-HASH.js gzip | 322 B | 322 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 10.1 kB | 10.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| _buildManifest.js gzip | 511 B | 511 B | ✓ |
| Overall change | 511 B | 511 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/tracing-bench-improvements | Change |
| - | - | - | - |
| index.html gzip | 539 B | 539 B | ✓ |
| link.html gzip | 551 B | 551 B | ✓ |
| withRouter.html gzip | 532 B | 532 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

</details>
Commit: d30643709475ac73a34770bb40339c94ab9b8cd6

**Comment by ijjk** [comment]
## Failing test suites 
Commit: d30643709475ac73a34770bb40339c94ab9b8cd6

**test/development/basic/hmr.test.ts** 
- basic HMR > Error Recovery > should recover after a bad return from the render function
- basic HMR > Error Recovery > should recover after undefined exported as default
- basic HMR > Error Recovery > should recover from errors in getInitialProps in client
- basic HMR > Error Recovery > should recover after an error reported via SSR

<details>
<summary>Expand output</summary>

  ● basic HMR › Error Recovery › should recover after a bad return from the render function

    TIMED OUT: /This is the about page/

      437 |
      438 |   if (hardError) {
    > 439 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content)
          |           ^
      440 |   }
      441 |   return false
      442 | }

      at Object.check (lib/next-test-utils.js:439:11)
      at Object.<anonymous> (development/basic/hmr.test.ts:558:9)

  ● basic HMR › Error Recovery › should recover after undefined exported as default

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `basic HMR Error Recovery should recover after undefined exported as default 1`

    - Snapshot  - 3
    + Received  + 1

       1 of 1 unhandled error
    - Server Error
    + Unhandled Runtime Error

      Error: The default export is not a React Component in page: "/hmr/about7"
    -
    - This error happened while generating the page. Any console logs will be displayed in the terminal window.

      600 |         await next.patchFile(aboutPage, aboutContent)
      601 |
    > 602 |         if (browser) {
          |                       ^
      603 |           await check(
      604 |             () => getBrowserBodyText(browser),
      605 |             /This is the about page/

      at Object.<anonymous> (development/basic/hmr.test.ts:602:23)

  ● basic HMR › Error Recovery › should recover from errors in getInitialProps in client

    TIMED OUT: /Hello/

      437 |
      438 |   if (hardError) {
    > 439 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content)
          |           ^
      440 |   }
      441 |   return false
      442 | }

      at Object.check (lib/next-test-utils.js:439:11)
      at Object.<anonymous> (development/basic/hmr.test.ts:659:8)

  ● basic HMR › Error Recovery › should recover after an error reported via SSR

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      701 |
      702 |         throw err
    > 703 |       } finally {
          |                  ^
      704 |         if (browser) {
      705 |           await browser.close()
      706 |         }

      at Object.<anonymous> (development/basic/hmr.test.ts:703:18)

</details>

File: bench/nested-deps/next.config.js [hunk 2]
```diff
@@ -2,6 +2,10 @@ module.exports = {
   eslint: {
     ignoreDuringBuilds: true,
   },
+  webpack(config) {
+    config.module.unsafeCache = true
```
Comment (sokra) [inline bench/nested-deps/next.config.js:6]: This is not needed, as it's now default

File: bench/nested-deps/package.json [hunk 3]
```diff
@@ -1,10 +1,11 @@
 {
   "scripts": {
     "prepare": "rm -rf components && mkdir components && node ./fuzzponent.js -d 2 -s 206 -o components",
-    "dev": "../../node_modules/.bin/next dev",
-    "build": "../../node_modules/.bin/next build",
-    "start": "../../node_modules/.bin/next start",
+    "dev": "NEXT_PRIVATE_LOCAL_WEBPACK5=1 ../../node_modules/.bin/next dev",
```
Comment (sokra) [inline bench/nested-deps/package.json:4]: This is not windows compatible. You may use `cross-env`

**Review by sokra — CHANGES_REQUESTED** [review]

**Pull request merged**